src/Backend/Form/EventListener/AddMetaSubscriber.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Backend\Form\EventListener;
+
+use Backend\Core\Engine\Model;
+use Backend\Form\Type\MetaType;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Use this class to add meta url generating to your symfony form.
+ *
+ * Example: $builder->addEventSubscriber(new AddMetaSubscriber(...));
+ */
+class AddMetaSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var string - The URL shown in the backend will need this "action" to be generated.
+     */
+    private $actionForUrl;
+
+    /**
+     * @var string - The field in the form where the URL should be generated for.
+     */
+    private $baseFieldName;
+
+    /**
+     * @var string - Name of the class or the container service id, f.e.: 'moduleForUrl.repository.item',
+     */
+    private $generateUrlCallbackClass;
+
+    /**
+     * @var string - Name of the public method which returns you the URL, f.e.: "getUrl"
+     */
+    private $generateUrlCallbackMethod;
+
+    /**
+     * @var array - Extra parameters you want to include in the AJAX call to get the URL, f.e.: ["getData.getLocale", "getForm.getParent.getParent.getData.getMenuEntityId"]
+     */
+    private $generateUrlCallbackParameterMethods;
+
+    /**
+     * @var string - The URL shown in the backend will need this "module" to be generated.
+     */
+    private $moduleForUrl;
+
+    public function __construct(
+        string $moduleForUrl,
+        string $actionForUrl,
+        string $baseFieldName = 'title',
```

Comment:
@jeroendesloovere you shouldn't do this, default parameters should be at the end, else they have no point at all since you would need to pass the baseFieldName anyway